google/resource_compute_target_tcp_proxy.go

```diff
@@ -0,0 +1,171 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+)
+
+func resourceComputeTargetTcpProxy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeTargetTcpProxyCreate,
+		Read:   resourceComputeTargetTcpProxyRead,
+		Delete: resourceComputeTargetTcpProxyDelete,
+		Update: resourceComputeTargetTcpProxyUpdate,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"backend_service": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"proxy_header": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NONE",
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"self_link": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceComputeTargetTcpProxyCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	proxy := &compute.TargetTcpProxy{
+		Name:    d.Get("name").(string),
+		Service: d.Get("backend_service").(string),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		proxy.Description = v.(string)
+	}
+
+	log.Printf("[DEBUG] TargetTcpProxy insert request: %#v", proxy)
+	op, err := config.clientCompute.TargetTcpProxies.Insert(
+		project, proxy).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating TargetTcpProxy: %s", err)
+	}
+
+	err = computeOperationWait(config, op, project, "Creating Target Tcp Proxy")
+	if err != nil {
+		return err
+	}
+
+	d.SetId(proxy.Name)
```

Comment:
Should probably set this to proxy.Id. 

Can two separate proxies have the same name?

Reply:
At least through the UI you can not have more than one with the same name. I'd think I'd leave it with name which is also in line with the others target resources